src/thunderbots/software/backend_output/grsim/motion_controller.cpp

```diff
@@ -0,0 +1,206 @@
+    if(delta_time <= 0) {
+        robot_velocities.linear_velocity = robot.velocity();
+        robot_velocities.angular_velocity = robot.angularVelocity();
+    }
+    else {
+
+        robot_velocities.linear_velocity = MotionController::determineLinearVelocity(
+                robot, dest, desired_final_speed, delta_time);
+        robot_velocities.angular_velocity = MotionController::determineAngularVelocity(
+                robot, desired_final_orientation, delta_time);
+    }
+    
+    return robot_velocities;
+}
+
+AngularVelocity MotionController::determineAngularVelocity(
+    const Robot robot, const Angle desired_final_orientation, const double delta_time)
+{
+    // boolean value if the robot can reach it's destination at target speed based on
+    // angular MAX acceleration
+    bool can_stop_rotate_in_time;
+
+    AngularVelocity robot_angular_velocity;
+
+    // rotation used for constant angular acceleration calculations
+    const Angle angle_to_dest = (robot.orientation() - desired_final_orientation);
+    AngularVelocity delta_angular_speed;
+    AngularVelocity expected_final_ang_speed;
+
+    // check negative sqrt case for angular speed
+    if (pow(robot.angularVelocity().toRadians(), 2) <=
+        fabs(2 * ROBOT_MAX_ANG_ACCELERATION * angle_to_dest.toRadians()))
+    {
+        expected_final_ang_speed = AngularVelocity::ofRadians(
+            -1 * sqrt(fabs(2 * ROBOT_MAX_ANG_ACCELERATION * angle_to_dest.toRadians()) -
+                      pow(robot.angularVelocity().toRadians(), 2)));
+        can_stop_rotate_in_time = true;
+    }
+    else
+    {
+        expected_final_ang_speed = AngularVelocity::ofRadians(
+            sqrt(pow(robot.angularVelocity().toRadians(), 2) -
+                 2 * ROBOT_MAX_ANG_ACCELERATION * angle_to_dest.toRadians()));
+        can_stop_rotate_in_time =
+            expected_final_ang_speed <= AngularVelocity::ofRadians(0.0);
+    }
+
+    // if the robot can stop rotating in time
+    if (can_stop_rotate_in_time)
+    {
+        // if the final expected angle is less than the desired angle then angularly
+        // accelerate
+        if (expected_final_ang_speed < AngularVelocity::ofRadians(0.0))
+        {
+            // angularly accelerate
+            delta_angular_speed =
+                AngularVelocity::ofRadians(ROBOT_MAX_ANG_ACCELERATION * delta_time);
+        }
+
+        // if not then maintain angular velocity
+        else
+        {
+            delta_angular_speed = AngularVelocity::ofRadians(0.0);
+        }
+    }
+
+    // if the robot can't stop rotating in time then angular decelerate
+    else
+    {
+        delta_angular_speed =
+            AngularVelocity::ofRadians(-(ROBOT_MAX_ANG_ACCELERATION * delta_time));
+    }
+
+    robot_angular_velocity =
+        robot.angularVelocity() +
+        delta_angular_speed;  // calculate new angular velocity based on the current robot
+    // ang. velocity and delta ang. velocity
```

Comment:
nit: For consistency, do not abbreviate angular here.

Reply:
Motion controller changed and this is no longer an issue.